controller/api/destination/watcher/ip_watcher.go

```diff
@@ -146,11 +169,37 @@ func (iw *IPWatcher) GetSvcID(clusterIP string) (*ServiceID, error) {
 }
 
 // GetPod returns the pod that corresponds to an IP address if one exists.
-func (iw *IPWatcher) GetPod(podIP string) (*corev1.Pod, error) {
-	objs, err := iw.k8sAPI.Pod().Informer().GetIndexer().ByIndex(podIPIndex, podIP)
+func (iw *IPWatcher) GetPod(podIP string, port uint32) (*corev1.Pod, error) {
+	addr := fmt.Sprintf("%s:%d", podIP, port)
+	hostIPPods, err := iw.getIndexedPods(hostIPIndex, addr)
+	if err != nil {
+		return nil, status.Error(codes.Unknown, err.Error())
+	}
+	// If there is exactly 1 pod found then it is returned. More than 1 pod is
+	// not handled because that would violate the k8s contract of unique
+	// hostIP:hostPort addresses.
+	if len(hostIPPods) == 1 {
+		return hostIPPods[0], nil
+	}
+	podIPPods, err := iw.getIndexedPods(podIPIndex, podIP)
 	if err != nil {
 		return nil, status.Error(codes.Unknown, err.Error())
 	}
+	if len(podIPPods) == 1 {
+		return podIPPods[0], nil
+	}
+	if len(podIPPods) > 1 {
```

Comment:
ideally we should make reasoning watertight enough that we know exactly what scenarios would lead us to this point. (or to give us confidence that it's impossible to get here)

Reply:
So with the change to the pod IP indexer exlcluding host network pods, this should be as watertight as the k8s contract allows it to be. If we get here, then that's an error case.